docs/how-to-build-with-ddn/with-oracle.mdx

```diff
@@ -0,0 +1,430 @@
+---
+sidebar_position: 6
+sidebar_label: With Oracle
+description: "Learn the basics of Hasura DDN and how to get started with an Oracle database."
+keywords:
+  - hasura ddn
+  - graphql api
+  - getting started
+  - guide
+  - oracle
+---
+
+import Prereqs from "@site/docs/_prereqs.mdx";
+
+# Get Started with Hasura DDN and Oracle
+
+## Overview
+
+This tutorial takes about twenty minutes to complete. You'll learn how to:
+
+- Set up a new Hasura DDN project
+- Connect it to an Oracle database
+- Generate Hasura metadata
+- Create a build
+- Run your first query
+- Create relationships
+
+Additionally, we'll familiarize you with the steps and workflows necessary to iterate on your API.
+
+This tutorial assumes you're starting from scratch. We'll use an AWS instance of Oracle and connect it to Hasura, but
+you can easily follow the steps if you already have data seeded; Hasura will never modify your source schema.
+
+<Prereqs />
+
+## Tutorial
+
+### Step 1. Authenticate your CLI
+
+```sh title="Before you can create a new Hasura DDN project, you need to authenticate your CLI:"
+ddn auth login
+```
+
+This will launch a browser window prompting you to log in or sign up for Hasura DDN. After you log in, the CLI will
+acknowledge your login, giving you access to Hasura Cloud resources.
+
+### Step 2. Scaffold out a new local project
+
+```sh title="Next, create a new local project:"
+ddn supergraph init my-project && cd my-project
+```
+
+Once you move into this directory, you'll see your project scaffolded out for you. You can view the structure by either
+running `ls` in your terminal, or by opening the directory in your preferred editor.
+
+### Step 3. Initialize your Oracle connector
+
+```sh title="In your project directory, run:"
+ddn connector init my_oracle -i
+```
+
+From the dropdown, start typing `oracle` and hit enter to accept the default port. Then, provide the following
+values:
+
+**JDBC URL**
+
+```plaintext
+jdbc:oracle:thin:@local.hasura.dev:1521/XEPDB1?user=example&password=mypassword
+```
+
+For Schemas, hit enter to use the database provided in the JDBC URL.
+
+### Step 4. Start the local Oracle container
+
+```sh title="Begin by creating a compose file for the Oracle service:"
+touch app/connector/my_oracle/compose.oracle.yaml
+```
+
+```yaml title="Then, open the file and add the following:"
+services:
+  oracle:
+    image: gvenzl/oracle-xe:21.3.0-slim
+    container_name: oracle-server
+    restart: always
+    ports:
+      - "1521:1521"
+    environment:
+      ORACLE_PASSWORD: mypassword
+      APP_USER: example
+      APP_USER_PASSWORD: mypassword
+      TARGET_PDB: XEPDB1
+```
+
+```sh title="Run the container:"
+docker compose -f app/connector/my_oracle/compose.oracle.yaml up -d
+```
+
+```sh title="Use the SQL*PLUS shell to start the SQL prompt:"
+docker exec -it oracle-server sqlplus example/mypassword@local.hasura.dev:1521/XEPDB1
+```
+
```

Comment:
Can we add a line saying the user is `example` and the password: `mypassword` as per the yaml.